src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp

```diff
@@ -32,12 +32,17 @@ struct fully_connected_onednn : typed_primitive_onednn_impl<fully_connected> {
     dnnl::memory::data_type _ds_data_type;
```

Comment:
random spot)
The OneDNN spec says `all tensors must have the same number of dimensions.` https://uxlfoundation.github.io/oneDNN/dev_guide_matmul.html

This ticket title says 4d activation and 2d weights. Then we are not providing input to onednn properly. Could you check and share the onednn verbose log?